libqtile/widget/memory.py

```diff
@@ -39,27 +39,33 @@ class Memory(base.ThreadedPollText):
     SwapFree: Returns amount of swap free
     SwapUsed: Returns amount of swap in use
 """
+
     orientations = base.ORIENTATION_HORIZONTAL
     defaults = [
-        ("fmt", "{MemUsed}M/{MemTotal}M", "Formatting for field names.")
+        ("format", "{MemUsed}M/{MemTotal}M", "Formatting for field names."),
+        ("update_interval", 1.0, "Update interval for the Memory"),
     ]
 
     def __init__(self, **config):
-        super().__init__(**config)
+        base.ThreadedPollText.__init__(self, **config)
```

Comment:
What is the reason for this change? Isn't it semantically the same?

Reply:
It's not, because `super().__init__` refers to `_TextBox.__init__` (iirc) and we need to init via `ThreadedPollText.__init__` or it crashes. I tried both changes individually and both crashes the WM.

Reply:
Well, I'm a bit lost here.
- `ThreadedPollText` does not redefine `__init__`, so it must use its parent one, `InLoopPollText.__init__`, which itself calls `_TextBox.__init__`
- `super()` should do the same because it goes up one level, using the direct parent's `__init__` which is `ThreadedPollText.__init__`, etc.

But maybe there's things happening in `super()` that I don't know.

Anyway, what do you mean by both changes crash the WM? Which changes?

Reply:
Me too, it raised some error message earlier but now it worked just fine. I added a commit to change it back to `super().__init__`

I meant that changing `format` back to `fmt` crashes and using `super().__init__` crashes bot individually. But I was wrong about the super.

Reply:
OK thanks for the explanation :slightly_smiling_face: